core/src/main/java/org/apache/accumulo/core/client/admin/PluginConfig.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.client.admin;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ *
+ * @since 2.1.0
+ */
+abstract class PluginConfig<T extends PluginConfig<T>> {
```

Comment:
It seems like there is some overlap in this class with ```PluginEnvironment``` which was also introduced in 2.1.0.  Any thoughts on if these could be consolidated into one entity?

Reply:
I created this to avoid duplicating code between a few classes.  It may be better to just not have the type and duplicate the code.

Reply:
In b764df9 I dropped some other classes and kept this class.  The class has a very different purpose than PluginEnv.  Its allows a user to pass (on the client side) a class name and options as configuration to the server side where the plugin will be instantiated and used.  PluginEnv is something a plugin instantiated on the server side might be passed.